src/cats/SlideAnimation.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from "react";
+import clsx from "clsx";
+import styles from "./SlideAnimation.module.css";
+
+enum Direction {
+  Previous,
+  Next,
+}
+
+type SlideAnimationPropsType = {
+  child: JSX.Element;
+  direction: Direction;
+};
+
+function SlideAnimation({ child, direction }: SlideAnimationPropsType) {
+  const [slideChildren, setSlideChildren] = useState<Array<JSX.Element>>([]);
+
+  useEffect(() => {
+    setSlideChildren((prevChildren: Array<JSX.Element>) => {
+      const prevChild = prevChildren.slice(
```

Comment:
consider, now that it is supported in newer browser:

```tsx
const prevChild = prevChildren.at(direction === Direction.Next ? -1 : 0);
```

Or for better support:

```tsx
const prevChild = prevChildren[direction === Direction.Next ? prevChidlren.length - 1 : 0];
```

Or the FB way:

```tsx
const prevChild = direction === Direction.Next ? last(prevChildren) : first(prevChildren);
```